modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/Environment.scala

```diff
@@ -204,7 +204,7 @@ object Environment {
       getRegionFromConfig(file).orElse(getRegion),
       cloud,
       acceptInvalid,
-      Some(() => metadata.submit.compile.drain)
+      metadata.submit.interruptAfter(1.minute)
```

Comment:
I'm still not at ease with the fact that the shutdown will wait for an infinite stream to be interrupted 1 minute later (unless I'm missing something?). What about having `def flush: F[Unit]` [in here](https://github.com/snowplow/enrich/blob/3.1.2/modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/io/experimental/Metadata.scala#L46-L50) that sends what's in `MetadataEventsRef.snapshot` when called ?

Reply:
Why infinite? We take a snapshot, then for all events we execute HTTP requests. This means a finite stream. The interruption will stop if metadata submitting takes longer than that which means that upstream endpoint is not responding and metadata cannot be submitted. I think capping that makes sense.

Reply:
I managed to get the integration test to pass by changing [this line](https://github.com/snowplow/enrich/blob/master/modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/io/experimental/Metadata.scala#L87) to `Stream.empty[F]` instead of `Stream.never[F]`.  No other change needed.  I think semantically this might be the correct fix: if it's a `noop` implementation of metadata then `submit` should return immediately, not never.